### ISSUE TITLE
Adding TrashCan causes IceTray to hang indefinitely (Trac #2368)

### ISSUE THREAD

**Issue by icecube-trac** [icecube-trac/tix2#2360]
This code hangs forever after the first frame has reached the TrashCan

```text
from icecube import icetray, dataio
tray = I3Tray.I3Tray()
tray.Add("I3InfiniteSource")
tray.Add("TrashCan")
tray.Execute(10)
```

Migrated from https://code.icecube.wisc.edu/ticket/2368
```json
{
    "status": "closed", 
    "changetime": "2019-10-23T12:20:01", 
    "description": "This code hangs forever after the first frame has reached the TrashCan\n{{{\nfrom icecube import icetray, dataio\ntray = I3Tray.I3Tray()\ntray.Add(\"I3InfiniteSource\")\ntray.Add(\"TrashCan\")\ntray.Execute(10)\n}}}\n", 
    "reporter": "chaack", 
    "cc": "", 
    "resolution": "fixed", 
    "_ts": "1571833201520712", 
    "component": "combo core", 
    "summary": "Adding TrashCan causes IceTray to hang indefinitely", 
    "priority": "major", 
    "keywords": "", 
    "time": "2019-10-23T07:48:44", 
    "milestone": "", 
    "owner": "jvansanten", 
    "type": "defect"
}
```

**Comment by icecube-trac** [comment]
Trac update at 20191023T07:48:54: chaack changed summary from "Adding TrashCan causes IceTray to hand indefinitely" to "Adding TrashCan causes IceTray to hang indefinitely"

**Comment by icecube-trac** [comment]
Trac update at 20191023T07:51:00: jvansanten commented:

> This is likely related to #1990.

**Comment by icecube-trac** [comment]
Trac update at 20191023T08:23:19: 
* jvansanten changed owner from "" to "jvansanten"
* jvansanten commented:

> In changeset 176356:
> 
> ```text
> #!CommitTicketReference repository="IceCube" revision="176356"
> Prevent TrashCan from looping indefinitely
> 
> TrashCan may not be necessary, but if it actually does nothing then the
> previous module's outbox will never drain, and tray will run forever.
> Fixes #2360.
> ```


* jvansanten changed resolution from "" to "fixed"
* jvansanten changed status from "new" to "closed"

**Comment by icecube-trac** [comment]
Trac update at 20191023T08:26:44: jvansanten commented:

> For completeness, the regression was in changeset 175991.

**Comment by icecube-trac** [comment]
Trac update at 20191023T12:20:01: eganster commented:

> In changeset 176362:
> 
> ```text
> #!CommitTicketReference repository="IceCube" revision="176362"
> merges r176356 to fix #2360 in the snobo branch
> ```
